File: python/2610. Convert an Array Into a 2D Array With Conditions.py
from ast import List


class Solution:
    def findMatrix(self, nums: List[int]) -> List[List[int]]:
        hash_map = dict()
        size = 1
        for num in nums:
            if num in hash_map:
                temp = hash_map[num]
                hash_map[num] = temp+1
                size = max(size,temp+1)
            else:
                hash_map[num] = 1
        ret = [[] for i in range(size)]
        for k,v in hash_map.items():
            for i in range(v):
                ret[i].append(k)
        return ret
        

File: java/2610. Convert an Array Into a 2D Array With Conditions.java
import java.util.*;

class Solution {
    public List<List<Integer>> findMatrix(int[] nums) {
        HashMap<Integer,Integer> hm = new HashMap<>();
        List<List<Integer>> ans = new ArrayList<>();
        int maxSize = 1;
        for(int num: nums){
            hm.put(num,hm.getOrDefault(num,0)+1);
            maxSize = Math.max(maxSize,hm.get(num));
        }
        for(int i = 0; i < maxSize; i++){
            List<Integer> temp = new ArrayList<>();
            ans.add(temp);
        }
        for(int key: hm.keySet()){
            int val = hm.get(key);
            for(int i = 0; i < val; i++){
                ans.get(i).add(key);
            }
        }
        return ans;
    }
}